java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/RecordFormattingTest.java

```diff
@@ -0,0 +1,200 @@
+    private static final List<String> EXTRA_OPTIONS = new ArrayList<>();
+
+    /**
+     * Creates a new instance of FormatingTest
+     */
+    public RecordFormattingTest(String name) {
+        super(name);
+    }
+
+    public static NbTestSuite suite() {
+        NbTestSuite suite = new NbTestSuite();
+        suite.addTestSuite(RecordFormattingTest.class);
+        return suite;
+    }
+
+    String content
+            = """
+              record Student(int id,String lastname,String firstname) implements Serializable {
+              int m(int x){ 
+              return id+x;
+              }
+              } // should stay flush to left margin
+              """;
+
+    public void test7043NoSpaces() throws Exception {
+        run7043("""
+                  record Student(int id, String lastname, String firstname) implements Serializable {
+                      int m(int x) {
+                          return id + x;
+                      }
+                  } // should stay flush to left margin
+                  """, false);
+    }
+
+    public void test7043WithSpaces() throws Exception {
+        run7043("""
+                  record Student( int id, String lastname, String firstname ) implements Serializable {
+                      int m( int x ) {
+                          return id + x;
+                      }
+                  } // should stay flush to left margin
+                  """, true);
+    }
+
+    /**
+     * The java formatter indents the final brace '}' of a record to far to the
+     * right, when the setting spaceWithinMethodDeclParens is True AND the
+     * record header defines parameters. This
+     *
+     * @throws Exception for some reason
+     */
+    // copied from testSealed
+    public void run7043(String golden, boolean spacesInMethodDecl) throws Exception {
+        try {
+            SourceVersion.valueOf("RELEASE_17"); //NOI18N
+        } catch (IllegalArgumentException ex) {
+            //OK, no RELEASE_15, skip test
+            return;
+        }
+        testFile = new File(getWorkDir(), "Test.java");
+        TestUtilities.copyStringToFile(testFile, "");
+        FileObject testSourceFO = FileUtil.toFileObject(testFile);
+        DataObject testSourceDO = DataObject.find(testSourceFO);
+        EditorCookie ec = (EditorCookie)testSourceDO.getCookie(EditorCookie.class);
+        Preferences preferences = MimeLookup.getLookup(JavaTokenId.language().mimeType()).lookup(Preferences.class);
+        preferences.putBoolean("spaceWithinMethodDeclParens", spacesInMethodDecl);
+        preferences.putInt("blankLinesAfterClassHeader", 0);
+
+        final Document doc = ec.openDocument();
+        doc.putProperty(Language.class, JavaTokenId.language());
+//        System.err.println("\n============== start test ========================");
+        System.err.println("golden length = " + golden.trim().length());
+        System.err.println("content length = " + content.trim().length());
```

Comment:
no println in tests please. We have hours of tests and have to keep the output manageable